fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/RecordValidationTest.java

```diff
@@ -0,0 +1,238 @@
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, formatVersion, hook);
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            validateRecord(store, result.get(0), null);
+            validateRecord(store, result.get(1), null);
+            context.commit();
+        }
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsMissingRecord [splitLongRecords = {0}], formatVersion = {1}")
+    @MethodSource("splitAndVersion")
+    public void testValidateRecordsMissingRecord(boolean splitLongRecords, int formatVersion) throws Exception {
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, formatVersion, hook);
+        // Delete a record
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            store.deleteRecord(result.get(0).getPrimaryKey());
+            commit(context);
+        }
+        // Validate by primary key
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            validateRecord(store, result.get(0), FDBRecordStoreBase.RecordValidationOptions.RECORD_EXISTS);
+            validateRecord(store, result.get(1), null);
+            context.commit();
+        }
+    }
+
+    @Nonnull
+    public static Stream<Arguments> splitNumberAndFormatVersion() {
+        return Stream.of(0, 1, 2)
+                .flatMap(split -> formatVersions()
+                        .map(version -> Arguments.of(split, version)));
+    }
+
+    @ParameterizedTest(name = "testValidateRecordsMissingSplit [splitNumber = {0}, formatVersion = {1}]")
+    @MethodSource("splitNumberAndFormatVersion")
+    public void testValidateRecordsMissingSplit(int splitNumber, int formatVersion) throws Exception {
+        boolean splitLongRecords = true;
+        // unsplit (short) records have split #0; split records splits start at #1
+        // Use this number to decide which record to operate on
+        int recordNumber = (splitNumber == 0) ? 0 : 1;
+
+        final RecordMetaDataHook hook = getRecordMetaDataHook(splitLongRecords);
+        List<FDBStoredRecord<Message>> result = saveRecords(splitLongRecords, formatVersion, hook);
+        // Delete a split
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = openSimpleRecordStore(context, hook, formatVersion);
+            // If operating on the short record, #0 is the only split
+            // If operating on the long record, splits can be 1,2,3
```

Comment:
This should probably explain why you aren't testing with `splitNumber=3`

Reply:
Done.